source/common/common/scalar_to_byte_vector.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <inttypes.h>
+
+#include <vector>
+
+namespace Envoy {
+template <typename T> void pushScalarToByteVector(T val, std::vector<uint8_t>& bytes) {
+  uint8_t* byte_ptr = reinterpret_cast<uint8_t*>(&val);
+  for (unsigned int byte_index = 0; byte_index < sizeof val; byte_index++) {
```

Comment:
Nit: Envoy style prefers fixed type sizes, e.g. `uint32_t` here.

Reply:
Done in https://github.com/envoyproxy/envoy/pull/4973/commits/d6583b94b1d64a73fa18c88f0f49f220e7a989ce